pkg/cli/admin/catalog/mirror.go

```diff
@@ -0,0 +1,281 @@
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/klog"
+	"k8s.io/kubectl/pkg/util/templates"
+	"sigs.k8s.io/yaml"
+
+	imgextract "github.com/openshift/oc/pkg/cli/image/extract"
+	imgmirror "github.com/openshift/oc/pkg/cli/image/mirror"
+)
+
+var (
+	mirrorLong = templates.LongDesc(`
+			Mirrors the contents of a catalog into a registry.
+
+			This command will pull down an image containing a catalog database, extract it to disk, query it to find
+            all of the images used in the manifests, mirrors them to a target registry.
+
+			By default, the database is extracted to a temporary directory, but can be saved locally via flags.
+
+			An ImageContentSourcePolicy is written to a file for application to a cluster with access to the target 
+			registry. This will configure the cluster to pull from the mirrors instead of the locations referenced in
+			the operator manifests.
+
+			A mapping.txt file is also created that is compatible with "oc image mirror". This may be used to further
+			customize the mirroring configuration, but should not be needed in normal circumstances.
+		`)
+)
+
+type MirrorCatalogOptions struct {
+	*mirror.IndexImageMirrorerOptions
+	DryRun       bool
+	ManifestOnly bool
+	DatabasePath string
+
+	SourceRef reference.DockerImageReference
+	Dest      string
+}
+
+func (o *MirrorCatalogOptions) Complete(cmd *cobra.Command, args []string) error {
+	if len(args) < 2 {
+		return fmt.Errorf("must specify source and dest")
+	}
+	src := args[0]
+	dest := args[1]
+
+	srcRef, err := reference.Parse(src)
+	if err != nil {
+		return err
+	}
+	o.SourceRef = srcRef
+	o.Dest = dest
+
+	if o.ManifestDir == "" {
+		o.ManifestDir = o.SourceRef.Name + "-manifests"
+	}
+
+	if err := os.MkdirAll(o.ManifestDir, os.ModePerm); err != nil {
+		return err
+	}
+
+	if o.DatabasePath == "" {
+		tmpdir, err := ioutil.TempDir("", "")
+		if err != nil {
+			return err
+		}
+		o.DatabasePath = "/:" + tmpdir
+	} else {
+		dir := strings.Split(o.DatabasePath, ":")
+		if len(dir) < 2 {
+			return fmt.Errorf("invalid path")
+		}
+		if err := os.MkdirAll(path.Dir(dir[1]), os.ModePerm); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (o *MirrorCatalogOptions) Validate() error {
+	if o.DatabasePath == "" {
+		return fmt.Errorf("must specify path for database")
+	}
+	if o.ManifestDir == "" {
+		return fmt.Errorf("must specify path for manifests")
+	}
+	return nil
+}
+
+func NewMirrorCatalog(streams genericclioptions.IOStreams) *cobra.Command {
+	o := MirrorCatalogOptions{
+		IndexImageMirrorerOptions: mirror.DefaultImageIndexMirrorerOptions(),
+	}
+
+	cmd := &cobra.Command{
+		Use:   "mirror",
+		Short: "mirror an operator-registry catalog",
+		Long:  mirrorLong,
+
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if debug, _ := cmd.Flags().GetBool("debug"); debug {
+				logrus.SetLevel(logrus.DebugLevel)
```

Comment:
Nope.

Reply:
The libraries I'm calling use `logrus`. Is there a restriction around calling libraries that use different log frameworks?

Reply:
I removed this for now - I can look into klog for the dependency if we need to see those in the future.